pkg/filters/labels.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package filters
+
+import (
+	"context"
+	"fmt"
+
+	k8sLabels "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels"
+	v1 "github.com/cilium/hubble/pkg/api/v1"
+	hubbleFilters "github.com/cilium/hubble/pkg/filters"
+	"github.com/cilium/tetragon/api/v1/tetragon"
+)
+
+// FilterByLabelSelectors returns a FilterFunc. The FilterFunc returns true if and only if any of the
+// specified selectors select the event. The caller specifies how to extract labels from the event.
+func FilterByLabelSelectors(labelSelectors []string) (hubbleFilters.FilterFunc, error) {
+	selectors := make([]k8sLabels.Selector, 0, len(labelSelectors))
+	for _, selector := range labelSelectors {
+		s, err := k8sLabels.Parse(selector)
+		if err != nil {
+			return nil, err
+		}
+		selectors = append(selectors, s)
+	}
+	return func(ev *v1.Event) bool {
+		process := GetProcess(ev)
+		if process == nil || process.Pod == nil {
+			return false
```

Comment:
I guess this means that if user specified 'key != label', which according to https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors is legit, an event with no pod information will not match. I think that's (probaby) fine, but maybe we should document it?

(see other comment)

Reply:
yeah that's an interesting point, let me think about this a bit and i'll document it 💭